projects/novo-elements/src/elements/data-table/cell-headers/data-table-header-cell.component.ts

```diff
@@ -326,12 +368,79 @@ export class NovoDataTableCellHeader<T> implements IDataTableSortFilter, OnInit,
   public cancel(): void {
     this.multiSelectedOptions = this.filter ? [...this.filter] : [];
     this.dropdown.closePanel();
+    this.clearOptionFilter();
   }
 
   public filterMultiSelect(): void {
-    let actualFilter = this.multiSelectedOptions.length > 0 ? [...this.multiSelectedOptions] : undefined;
-    this.filterData(actualFilter);
-    this.dropdown.closePanel();
+    if ((this.multiSelectedOptions.length === 0 && !this.filter) || this.multiSelectedOptions === this.filter) {
+      this.multiSelectHasVisibleOptions() && this.dropdown ? (this.error = true) : null;
+    } else {
+      this.clearOptionFilter();
+      let actualFilter = this.multiSelectedOptions.length > 0 ? [...this.multiSelectedOptions] : undefined;
+      this.filterData(actualFilter);
+      this.dropdown.closePanel();
+    }
+  }
+
+  public multiSelectOptionFilter(optionFilter: string) {
+    this.multiSelectedOptionIsHidden.forEach((record) => {
+      if (record.option) {
+        record.hidden = !(
+          this.getOptionText(record.option)
+            .toLowerCase()
+            .startsWith(optionFilter.toLowerCase()) || this.isSelected(record.option, this.multiSelectedOptions)
+        );
+      }
+    });
+  }
+
+  public multiSelectOptionIsHidden(option: string | IDataTableColumnFilterOption): boolean {
+    return this.multiSelectedOptionIsHidden.find((record) => record.option === option).hidden;
+  }
+
+  public multiSelectHasVisibleOptions(): boolean {
+    return this.multiSelectedOptionIsHidden.some((record) => !record.hidden);
+  }
+
+  private getOptionText(option: string | IDataTableColumnFilterOption): string {
+    if (typeof option !== 'object') {
```

Comment:
I think it'd be preferable to type narrow based on a property that would be in IDataTableColumnFilterOption instead.  like `'label' in object` since label is the only required property on that interface.

Reply:
Can't do an `in` check on object because it could be a string